app.py

```diff
@@ -188,16 +223,6 @@ def before_request_proxy():
     def before_request_receive_proxy():
         print ('DEBUG TEST - RECEIVE PROXIED REQUEST', request.method, request.url, session_id ())
 
-# HTTP -> HTTPS redirect
-# https://stackoverflow.com/questions/32237379/python-flask-redirect-to-https-from-http/32238093
-if os.getenv ('REDIRECT_HTTP_TO_HTTPS'):
-    @app.before_request
-    def before_request_https():
-        if request.url.startswith('http://'):
-            url = request.url.replace('http://', 'https://', 1)
-            # We use a 302 in case we need to revert the redirect.
-            return redirect(url, code=302)
```

Comment:
I'm curious to know the reason behind adding this initially. Doesn't heroku do redirects from http => https automatically anyway? Since Talisman forces HTTPS, I don't think this section is needed anymore, and actually, I'm not even sure it was needed previously, but there's very likely to be some context that I'm missing here.

Reply:
Good question. I'll leave this to @fpereiro to answer.

Reply:
Hi @lpmi-13 ! If I remember correctly, Heroku doesn't do the redirect from http to https automatically, so that's why I added that block. I distinctly remember testing this in `alpha` and seeing how this block enforced the redirect. I just tested `wget --save-headers --max-redirect=0 --output-document - http://hedycode.com` and it replies with a 302 redirecting to `https`, so I'm now pretty sure we need this :)

Reply:
ah okay...so even after enabled the HSTS header via talisman, on the first connection from a unique client, we'd still want this manual redirect. I suppose we could also use the preload list, though this would require registering and IIRC can't be easily undone if it happens to break anything.

Not sure how old this page's info is, but basically looks like what we had is the only way to do this. https://help.heroku.com/J2R1S4T8/can-heroku-force-an-application-to-use-ssl-tls

I'll go ahead and revert the removal of this block for this particular PR.

Reply:
actually...I got curious and deployed a very basic flask app with the default talisman config, and it looks like a request using http redirects automatically with a 302 to https.

code is at https://github.com/lpmi-13/flask-talisman-demo/blob/main/app.py
and deployed at http://flask-talisman-demo.herokuapp.com/

It looks like in the docs that talisman _both_ sets the strict transport header as well as returns a 302 to https on all requests. Is there a way we could test out talisman, but without that explicit redirect block?

Reply:
ha...and actually, the code in talisman looks like it's basically doing the exact same thing our current code does
https://github.com/GoogleCloudPlatform/flask-talisman/blob/master/flask_talisman/talisman.py#L211